addon/record-data.js

```diff
@@ -421,6 +421,17 @@ export default class M3RecordData {
       return key in this._attributes;
     }
   }
+  /**
+   * @param {string} key
+   * @returns {boolean}
+   */
+  getServerAttr(key) {
+    if (this._baseRecordData) {
+      return this._baseRecordData.getServerAttr(key);
+    } else {
+      return this._data[key];
```

Comment:
I think we might need to check explicitly for `null` otherwise boolean attributes that are set to false would be treated as non existent in the server. What do you think?

Reply:
Nice catch

Reply:
Will add this check to the call sites